src/TorchSharp/NN/AdaptiveAvgPool2D.cs

```diff
@@ -29,7 +29,29 @@ public static partial class Modules
         [DllImport ("LibTorchSharp")]
         extern static IntPtr THSNN_AdaptiveAvgPool2d_ctor (IntPtr psizes, int length, out IntPtr pBoxedModule);
 
-        static public AdaptiveAvgPool2D AdaptiveAvgPool2D (long[] kernelSize)
+        /// <summary>
+        /// Applies a 2D adaptive average pooling over an input signal composed of several input planes.
+        /// The output is of size H x W, for any input size.The number of output features is equal to the number of input planes.
+        /// </summary>
+        /// <param name="outputSize">The target output size of the image of the form H x H.</param>
+        /// <returns></returns>
+        static public AdaptiveAvgPool2D AdaptiveAvgPool2D((long, long) outputSize)
```

Comment:
`(long, long)` is not idiomatic I think - we currently aren't using tuples in this way for known-length integer vector parameters, nor would it be idiomatic replacement for two parameters

So I think use this to avoid inconsistency - if we decided to adopt tuples it should be done systematically

```
static public AdaptiveAvgPool2D AdaptiveAvgPool2D(long outputLength, long outputWidth)
```

Reply:
Hmmm it's a `long[]` here: https://github.com/xamarin/TorchSharp/pull/214/files#diff-ec3f83c280eb9ad9e93409d310c45340d5650be301e30845750cb40ffc9671bcR38

So either two parameters and three parameters in each, or a `long[]` in each - and whichever it is should be consistent with other modules.

Reply:
Or just remove this overload.

Reply:
This was an attempt to encode the PyTorch semantics, but to do so in a statically typed way. Here's what the PT docs have to say:
```
output_size – the target output size of the image of the form H x W. Can be a tuple (H, W) or a single H for a square image H x H. H and W can be either a int, or None which means the size will be the same as that of the input.
```

So, in Python, you would call the operator either with something like '720', or '(28,28)' -- with the overloads, you get the same in C#.

Reply:
But you are right, it's inconsistently not a tuple on the 3D module.

Reply:
Yes, please do this kind of API ergonomics as a systematic separate end-to-end activity, based off like-for-like python-C#-F# API usage examples.   C#, F# and .NET have numerous ways to get the ergonomic effects we need - some of these multiply the API in one or more dimensions (e.g. overloads), so the techniques have to be used carefully.  

Python also offers many different ergonomic options and which of these we allow needs to be considered carefully.  

From the F# perspective e.g. DiffSharp [this RFC](https://github.com/fsharp/fslang-design/blob/master/RFCs/FS-1092-anonymous-type-tagged-unions.md) is also related as a way to reduce the overload-dimensionality-expansion problem.

So in these PRs I'd recommend dealing with the core capabilities for native code first.

Reply:
Okay, I removed the overloads on the 2D version. When the CI is done, I'll merge.

Reply:
Cool. We certainly need excellent C# and F# ergonomics - just want to separate that as an activity we can reason about separately.

Reply:
That's fair.

Reply:
There may also be ergonomic technical tricks we can learn from Tensorflow.NET here